module/zfs/spa.c

```diff
@@ -1793,6 +1917,15 @@ spa_activate(spa_t *spa, spa_mode_t mode)
 	 */
 	spa->spa_upgrade_taskq = taskq_create("z_upgrade", 100,
 	    defclsyspri, 1, INT_MAX, TASKQ_DYNAMIC | TASKQ_THREADS_CPU_PCT);
+
+	if (shared_log != NULL || fnvlist_lookup_boolean(config,
+	    ZPOOL_CONFIG_IS_SHARED_LOG)) {
+		spa->spa_chain_map_taskq = taskq_create("z_chain_map", 100,
+		    defclsyspri, 1, INT_MAX, TASKQ_DYNAMIC |
+		    TASKQ_THREADS_CPU_PCT);
```

Comment:
I haven't red to the point of this taskq usage, but I wonder whether we really need 100% of CPU threads per pool for this?  And do I read it right that it is for both parent and children pools?

Reply:
I have a separate taskq for the client pools and the providers so that there would be less contention if you happen to have multiple client pools syncing at the same time you don't end up with a lot of waiting (since the taskqs are doing IOs, so they will end up blocking). The one for the shared log pool itself is actually only used during import, and could be destroyed after that if we want to save resources.

Reply:
Aside of resources it is really annoying sometimes to see a zillion extra threads.